Docs/sphinx_documentation/source/AMReX_Profiling_Tools.rst

```diff
@@ -107,59 +118,220 @@ You must at least instrument main(), i.e
       amrex::Initialize(argc,argv);
       BL_PROFILE_VAR("main()",pmain);
 
-      ...
+      <AMReX code block>
 
       BL_PROFILE_VAR_STOP(pmain);
       amrex::Finalize();
     }
 
-You can then instrument any of your functions
+or
 
 ::
 
-    void YourClass::YourFunction()
+    void main_main()
     {
-      BL_PROFILE_VAR("YourClass::YourFunction()",object_name);  // this name can be any string
+        BL_PROFILE("main()");
 
-      // your function code
+        <AMReX code block>
     }
 
-Note that you do not need to put BL_PROFILE_VAR_STOP because the profiler will
-go out of scope at the end of the function.
+    int main(...)
+    {
+        amrex::Initialize(argc,argv);
+        main_main();
+        amrex::Finalize();
+    }
+
+You can then instrument any of your functions, or code blocks. There are four general
+profiler macro types available:
+
+1) A scoped timer, :cpp:`BL_PROFILE`:
+----------------------------------------------------------------------------------------
+
+These timers generate their own object names, so they can't be controlled after being defined.
+However, they are the cleanest and easiest to work with in many situations. They time from
+the point where the macro is called until the end of the enclosing scope. This macro is ideal
+for timing an entire function. For example:
+
+::
+
+    void YourClass::YourFunction()
+    {
+      BL_PROFILE("YourClass::YourFunction()");   // Timer starts here.
+
+      < Your Function Code Block>
+
+    }    // <------ Timer goes out of scope here, calling stop and returning the function time.
+
+Note that all AMReX timers are scoped and will call "stop" when the corresponding object is destroyed.
+This macro is unique because it can *only* stop when it goes out of scope.
 
-For other timers within an already instrumented function, add:
+.. HACK check meaning of _only_ here.
```

Comment:
```suggestion
```

Reply:
Sorry, that my fault. I will clean it up.

Reply:
Got it.